crates/pallet-domains/src/lib.rs

```diff
@@ -386,68 +412,21 @@ mod pallet {
-                    // but a new one with receipts {1, 2} is received too, at this point, due to
-                    // these two extrinsics provides the same tag (`1`), the tx pool will check
-                    // if the latter one need to replace the previous one, if we don't set a higher
-                    // priority for the extrinsic with receipts {1, 2}, it will be simply rejected
-                    // as too low priority. However, it makes more sense to replace the previous
-                    // one in this case or to have a better way of handling these extrinsics.
-                    //
-                    // #2: `submit_transaction_bundle` with receipts {1}
-                    // #3: `submit_transaction_bundle` with receipts {1, 2}
-                    //
-                    // Now the unthoughtful priority is caculated based on the assumption that an extrinsic
-                    // providing more receipts has a higher priority.
-                    let additional_priority = signed_opaque_bundle
+                    let receipt_numbers = signed_opaque_bundle
                         .bundle
                         .receipts
                         .iter()
-                        .map(|r| r.primary_number.saturated_into::<TransactionPriority>())
-                        .sum::<TransactionPriority>();
-                    builder = builder.priority(TransactionPriority::MAX / 2 + additional_priority);
-
-                    let first_primary_number = signed_opaque_bundle
-                        .bundle
-                        .receipts
-                        .get(0)
-                        .expect("Receipts in a bundle after Block #1 must be non-empty as checked above; qed")
-                        .primary_number;
+                        .map(|r| r.primary_number)
+                        .collect::<Vec<_>>();
 
-                    // primary_number is ensured to be larger than the best execution chain chain
-                    // number above.
-                    //
-                    // No requires if it's the next expected execution chain number.
-                    let (_, best_number) = <ReceiptHead<T>>::get();
-                    if first_primary_number == best_number + One::one() {
-                        builder.build()
-                    } else {
-                        builder
-                            .and_requires(first_primary_number - One::one())
-                            .build()
-                    }
+                    ValidTransaction::with_tag_prefix("SubspaceSubmitBundle")
+                        .priority(TransactionPriority::MAX)
+                        .longevity(T::ConfirmationDepthK::get().try_into().unwrap_or_else(|_| {
+                            panic!("Block number always fits in TransactionLongevity; qed")
+                        }))
+                        .and_provides(receipt_numbers)
```

Comment:
Do we need `provides` now though? It'll still prevent inclusion of multiple bundles containing the same/similar set of ERs and AFAIK we do want to have multiple bundles concurrently (at least eventually).